src/main/java/edu/usf/cutr/gtfsrtvalidator/api/resource/GtfsRtFeed.java

```diff
@@ -330,6 +354,84 @@ public Response getIterationDetails(
         return Response.ok(gtfsRtFeedIterationModel).build();
     }
 
+    // Returns past session details for a particular clientId.
+    @GET
+    @Path("/{clientId : .*}/pastSessions")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getSessionData(
+            @PathParam("clientId") String clientId) {
+
+        if (clientId.isEmpty()) {
+            String uuid = UUID.randomUUID().toString();
+            JsonObject json = new JsonObject();
+            json.addProperty("clientId", uuid);
+            return Response.ok(json.toString()).build();
+        }
+
+        List<SessionModel> sessionModelList;
+        Session session = GTFSDB.initSessionBeginTrans();
+        long timeDiff;
+        int rowId = 0;
+
+        sessionModelList = session.createQuery(" FROM SessionModel" +
+                " WHERE userId = '" + clientId + "'").list();
+
+        Iterator iterator = sessionModelList.listIterator();
+        SessionModel eachSessionModel;
+        DateFormat timeFormat = new SimpleDateFormat("MM/dd/yyyy HH:mm:ss");
+        while(iterator.hasNext()) {
+            eachSessionModel = (SessionModel) iterator.next();
+
+            eachSessionModel.setRowId(++rowId);
+
+            eachSessionModel.setStartTimeFormat(timeFormat.format(eachSessionModel.getSessionStartTime()));
+            eachSessionModel.setEndTimeFormat(timeFormat.format(eachSessionModel.getSessionEndTime()));
+
+            timeDiff = eachSessionModel.getSessionEndTime() - eachSessionModel.getSessionStartTime();
+            timeFormat.format(timeDiff);
+            eachSessionModel.setTotalTime(getTotalTimeFormat(timeDiff));
+        }
+
+        GenericEntity<List<SessionModel>> pastSessionsList = new GenericEntity<List<SessionModel>>(sessionModelList) {
+        };
+        return Response.ok(pastSessionsList).build();
+    }
+
+    // Update sessionEndTime, errorCount and warningCount of a session.
+    @PUT
+    @Path("/{sessionId}/closeSession")
+    public void updateSessionData(
+            @PathParam("sessionId") int sessionId) {
+
+        SessionModel sessionModel = new SessionModel();
```

Comment:
I believe you don't actually need to instantiate a new `SessionModel` here - it can just be `SessionModel sessionModel;`, or just declare it as part of line 410 `SessionModel sessionModel = (SessionModel) session....`.